File: apps/android/app/src/main/kotlin/com/wordocious/app/ui/ProfileRestat.kt
package com.wordocious.app.ui

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AutoAwesome
import androidx.compose.material.icons.filled.Bolt
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.LocalFireDepartment
import androidx.compose.material.icons.filled.TrackChanges
import androidx.compose.material.icons.filled.TrendingUp
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wordocious.app.data.AuthService
import com.wordocious.app.data.MatchStatsService
import com.wordocious.app.data.StatsDeepService
import com.wordocious.app.ui.theme.WTheme

/**
 * Profile restructure cards (restat R1–R3) — ports snapshot-hero.tsx and the
 * new profile/page.tsx sections (and iOS ProfileRestat.swift): SnapshotHero
 * (lifetime stats + this-week strip in ONE card), the "top X% today" standing
 * strip, Opener Lab, Weekday Form, and the Daily Points trend (split out of
 * the old sweep-counts card — the counts' single home is now Records → You).
 */

// ── Snapshot hero ─────────────────────────────────────────────────────────────

/** Merges the old 4-card summary row + "This Week" recap into ONE card:
 *  lifetime headline stats up top, the week strip underneath, and a Pro
 *  upsell link for free users. */
@Composable
fun SnapshotHero(
    totalWins: Int,
    totalLosses: Int,
    currentStreak: Int,
    bestStreak: Int,
    dailyStreak: Int,
    bestDailyStreak: Int,
    gamesThisWeek: Int,
    level: Int,
    xpToNext: Int,
    isPro: Boolean,
    onGoPro: () -> Unit,
) {
    val totalGames = totalWins + totalLosses
    val winRate = if (totalGames > 0) Math.round(totalWins.toFloat() / totalGames * 100) else 0
    KitCard {
        Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            // F4: marquee numbers count up from 0 on first appear (Win Rate keeps "%").
            StatCell(Icons.Filled.EmojiEvents, "Wins", "$totalWins", color = Color(0xFF7C3AED), modifier = Modifier.weight(1f), countUp = totalWins)
            StatCell(Icons.Filled.TrackChanges, "Win Rate", "$winRate%", color = Color(0xFF2563EB), modifier = Modifier.weight(1f), countUp = winRate, countSuffix = "%")
            StatCell(Icons.Filled.Bolt, "Streak", "$currentStreak", sub = "Best: $bestStreak", color = WTheme.primary, modifier = Modifier.weight(1f), countUp = currentStreak)
            StatCell(Icons.Filled.LocalFireDepartment, "Daily", "$dailyStreak", sub = "Best: $bestDailyStreak", color = Color(0xFFF97316), modifier = Modifier.weight(1f), countUp = dailyStreak)
        }
        Spacer(Modifier.height(12.dp))
        Box(Modifier.fillMaxWidth().height(1.dp).background(WTheme.border))
        Spacer(Modifier.height(12.dp))
        Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            Icon(Icons.Filled.AutoAwesome, null, tint = WTheme.primary, modifier = Modifier.size(13.dp))
            Text("THIS WEEK", fontSize = 10.sp, fontWeight = FontWeight.Black, letterSpacing = 0.5.sp, color = Color(0xFF6D28D9))
            Text(
                "$gamesThisWeek ${if (gamesThisWeek == 1) "game" else "games"}",
                fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.text, maxLines = 1,
            )
            Spacer(Modifier.weight(1f))
            Icon(Icons.Filled.TrendingUp, null, tint = Color(0xFF2563EB), modifier = Modifier.size(13.dp))
            Row {
                Text("$xpToNext XP ", fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.text, maxLines = 1)
                Text("to Lvl ${level + 1}", fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.textMuted, maxLines = 1)
            }
        }
        if (!isPro) {
            Spacer(Modifier.height(10.dp))
            Box(Modifier.fillMaxWidth().height(1.dp).background(WTheme.border))
            Row(
                Modifier.fillMaxWidth().clickableNoRipple(onGoPro).padding(top = 10.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text("Unlock your full insights with Pro", fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.primary)
                Spacer(Modifier.weight(1f))
                Text("→", fontSize = 12.sp, fontWeight = FontWeight.Black, color = WTheme.primary)
            }
        }
    }
}

// ── Daily standing strip ──────────────────────────────────────────────────────

/** "You're in the top X% today · across N dailies" — where today's composite
 *  scores sit in the field. Hidden when the user hasn't played a daily today.
 *  [reloadToken] bumps refetch (daily completions change the standing). */
@Composable
fun DailyStandingStrip(reloadToken: Int = 0) {
    var standing by remember { mutableStateOf<StatsDeepService.DailyStanding?>(null) }
    LaunchedEffect(reloadToken) {
        AuthService.userId?.let { standing = StatsDeepService.todayDailyStanding(it) }
    }
    val s = standing ?: return
    Row(
        Modifier.fillMaxWidth().clip(RoundedCornerShape(14.dp))
            .background(Brush.linearGradient(listOf(Color(0xFFF5F3FF), Color(0xFFFCE7F3))))
            .border(1.5.dp, Color(0xFFE9D5FF), RoundedCornerShape(14.dp))
            .padding(horizontal = 16.dp, vertical = 10.dp),
        verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Icon(Icons.Filled.TrendingUp, null, tint = WTheme.primary, modifier = Modifier.size(14.dp))
        Row {
            Text("You're in the ", fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.text)
            Text("top ${s.topPercent}%", fontSize = 11.sp, fontWeight = FontWeight.Black, color = WTheme.primary)
            Text(" today", fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.text)
            Text(
                " · across ${s.modesCounted} ${if (s.modesCounted == 1) "daily" else "dailies"}",
                fontSize = 11.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.textMuted,
            )
        }
    }
}

// ── Opener Lab (basic) ────────────────────────────────────────────────────────

/** Favorite starting words + how they convert (win rate of games opened with
 *  each word). Free-tier card — the deep yield version lives in Deep Insights. */
@Composable
fun OpenerLabCard(playType: String = "solo") {
    var openers by remember { mutableStateOf<List<StatsDeepService.OpenerStat>>(emptyList()) }
    LaunchedEffect(playType) {
        openers = AuthService.userId?.let { StatsDeepService.openerStats(it, 5, playType) } ?: emptyList()
    }
    if (openers.isEmpty()) return
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        SectionHeader("Opener Lab", accent = Color(0xFF06B6D4))
        KitCard {
            Column(verticalArrangement = Arrangement.spacedBy(6.dp)) {
                openers.forEachIndexed { i, o ->
                    Row(
                        Modifier.fillMaxWidth().clip(RoundedCornerShape(10.dp)).background(WTheme.bg).padding(8.dp),
                        verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(10.dp),
                    ) {
                        Text("${i + 1}", fontSize = 10.sp, fontWeight = FontWeight.Black, color = WTheme.textMuted, modifier = Modifier.width(16.dp), textAlign = TextAlign.Center)
                        Text(o.word, fontSize = 14.sp, fontWeight = FontWeight.Black, letterSpacing = 1.2.sp, color = WTheme.text, modifier = Modifier.weight(1f))
                        Text("${o.count}×", fontSize = 10.sp, fontWeight = FontWeight.Bold, color = WTheme.textMuted)
                        Text(
                            "${o.winRate}% W", fontSize = 12.sp, fontWeight = FontWeight.Black,
                            color = if (o.winRate >= 50) Color(0xFF7C3AED) else Color(0xFFDC2626),
                            modifier = Modifier.width(48.dp), textAlign = TextAlign.End,
                        )
                    }
                }
            }
            Spacer(Modifier.height(8.dp))
            Text(
                "Win rate of games opened with each word",
                fontSize = 9.sp, fontWeight = FontWeight.Bold, color = WTheme.textMuted,
                modifier = Modifier.fillMaxWidth(), textAlign = TextAlign.Center,
            )
        }
    }
}

// ── Weekday form ──────────────────────────────────────────────────────────────

/** Win rate by day of week — highlights your best day (gold bar). */
@Composable
fun WeekdayFormCard(playType: String = "solo") {
    var days by remember { mutableStateOf<List<StatsDeepService.WeekdayFormDay>>(emptyList()) }
    LaunchedEffect(playType) {
        days = AuthService.userId?.let { StatsDeepService.weekdayForm(it, playType) } ?: emptyList()
    }
    if (days.none { it.played > 0 }) return
    val labels = listOf("S", "M", "T", "W", "T", "F", "S")
    val dayNames = listOf("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday")
    val maxPlayed = maxOf(1, days.maxOf { it.played })
    val best = days.filter { it.played >= 3 }
        .maxByOrNull { it.won.toDouble() / it.played }
    val hint = best?.let { "Best: ${dayNames[it.dow]} (${Math.round(it.won * 100.0 / it.played)}%)" }
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        SectionHeader("Weekday Form", accent = Color(0xFFF97316))
        ChartCard(title = "Win rate by day", hint = hint) {
            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(6.dp), verticalAlignment = Alignment.Bottom) {
                days.forEach { d ->
                    val rate = if (d.played > 0) d.won.toFloat() / d.played else 0f
                    Column(Modifier.weight(1f), horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.spacedBy(4.dp)) {
                        Text(
                            if (d.played > 0) "${Math.round(rate * 100)}%" else " ",
                            fontSize = 8.sp, fontWeight = FontWeight.Bold, color = WTheme.textMuted,
                        )
                        Box(Modifier.fillMaxWidth().height(44.dp), contentAlignment = Alignment.BottomCenter) {
                            val barMod = Modifier.fillMaxWidth()
                                .height(if (d.played == 0) 2.dp else (4.4f + rate * 39.6f).dp)
                                .clip(RoundedCornerShape(topStart = 3.dp, topEnd = 3.dp))
                            val alpha = if (d.played == 0) 1f else 0.5f + 0.5f * d.played / maxPlayed
                            when {
                                d.played == 0 -> Box(barMod.background(WTheme.border))
                                best?.dow == d.dow -> Box(barMod.background(Brush.verticalGradient(listOf(Color(0xFFFBBF24).copy(alpha = alpha), Color(0xFFF97316).copy(alpha = alpha)))))
                                else -> Box(barMod.background(Brush.verticalGradient(listOf(Color(0xFFA78BFA).copy(alpha = alpha), Color(0xFF7C3AED).copy(alpha = alpha)))))
                            }
                        }
                        Text(labels[d.dow], fontSize = 9.sp, fontWeight = FontWeight.ExtraBold, color = WTheme.textMuted)
                    }
                }
            }
        }
    }
}

// ── Daily points trend ────────────────────────────────────────────────────────

/** Points-per-day line (sweep/flawless days marked) — split out of the old
 *  sweep-counts card; the counts moved to Records → You (single home). */
@Composable
fun DailyPointsChartCard(points: List<MatchStatsService.DailyPointsPoint>) {
    if (points.size < 2) return
    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
        SectionHeader("Daily Points", accent = Color(0xFFEC4899))
        ChartCard(title = "Points per day", hint = "Last 30 days · ● sweep · ● flawless") {
            val maxV = maxOf(1, points.maxOf { it.totalPoints })
            Canvas(Modifier.fillMaxWidth().height(110.dp)) {
                val w = size.width; val h = size.height
                fun x(i: Int) = w * i / (points.size - 1)
                fun y(v: Int) = h - (h * v / maxV)
                val areaPath = androidx.compose.ui.graphics.Path().apply {
                    moveTo(0f, h)
                    points.forEachIndexed { i, p -> lineTo(x(i), y(p.totalPoints)) }
                    lineTo(w, h); close()
                }
                drawPath(areaPath, brush = Brush.verticalGradient(listOf(Color(0xFFA78BFA).copy(alpha = 0.3f), Color.Transparent)))
                val linePath = androidx.compose.ui.graphics.Path().apply {
                    points.forEachIndexed { i, p -> if (i == 0) moveTo(x(i), y(p.totalPoints)) else lineTo(x(i), y(p.totalPoints)) }
                }
                drawPath(linePath, Color(0xFF7C3AED), style = androidx.compose.ui.graphics.drawscope.Stroke(width = 3f))
                points.forEachIndexed { i, p ->
                    if (p.swept || p.flawless) {
                        drawCircle(
                            if (p.flawless) Color(0xFFF59E0B) else Color(0xFFEC4899),
                            radius = 5f, center = androidx.compose.ui.geometry.Offset(x(i), y(p.totalPoints)),
                        )
                    }
                }
            }
        }
    }
}

File: apps/ios/Wordocious/Sources/ProfileRestat.swift
import SwiftUI
import Charts
import WordociousCore

/// Profile restructure cards (restat R1–R3) — ports snapshot-hero.tsx and the
/// new profile/page.tsx sections: SnapshotHero (lifetime stats + this-week
/// strip in ONE card), the "top X% today" standing strip, Opener Lab, Weekday
/// Form, and the Daily Points trend (split out of the old sweep-counts card —
/// the counts' single home is now Records → You).

// MARK: - Snapshot hero

/// Merges the old 4-card summary row + "This Week" recap into ONE card:
/// lifetime headline stats up top, the week strip underneath, and a Pro
/// upsell link for free users.
struct SnapshotHero: View {
    let profile: Profile
    let gamesThisWeek: Int
    let isPro: Bool
    @State private var showPro = false

    var body: some View {
        let totalGames = profile.totalWins + profile.totalLosses
        let winRate = totalGames > 0 ? Int((Double(profile.totalWins) / Double(totalGames) * 100).rounded()) : 0
        let xpToNext = 1000 - (profile.xp % 1000)
        KitCard {
            VStack(spacing: 0) {
                HStack(spacing: 8) {
                    StatCell(icon: "trophy.fill", label: "Wins", value: "\(profile.totalWins)", color: Color(hex: 0x7C3AED), countUp: profile.totalWins)
                    StatCell(icon: "target", label: "Win Rate", value: "\(winRate)%", color: Color(hex: 0x2563EB), countUp: winRate, countSuffix: "%")
                    StatCell(icon: "bolt.fill", label: "Streak", value: "\(profile.currentStreak)", sub: "Best: \(profile.bestStreak)", color: Theme.primary, countUp: profile.currentStreak)
                    StatCell(icon: "flame.fill", label: "Daily", value: "\(profile.dailyLoginStreak)", sub: "Best: \(profile.bestDailyLoginStreak)", color: Color(hex: 0xF97316), countUp: profile.dailyLoginStreak)
                }
                HStack(spacing: 6) {
                    Image(systemName: "sparkles").font(.system(size: 13)).foregroundStyle(Theme.primary)
                    Text("THIS WEEK").font(Brand.font(10, .black)).tracking(0.5).foregroundStyle(Color(hex: 0x6D28D9))
                    Text("\(gamesThisWeek) \(gamesThisWeek == 1 ? "game" : "games")")
                        .font(Brand.font(11, .heavy)).foregroundStyle(Theme.textPrimary).lineLimit(1)
                    Spacer(minLength: 8)
                    Image(systemName: "chart.line.uptrend.xyaxis").font(.system(size: 13)).foregroundStyle(Color(hex: 0x2563EB))
                    (Text("\(xpToNext) XP ").font(Brand.font(11, .heavy)).foregroundColor(Theme.textPrimary)
                     + Text("to Lvl \(profile.level + 1)").font(Brand.font(11, .heavy)).foregroundColor(Theme.textMuted))
                        .lineLimit(1)
                }
                .padding(.top, 12)
                .overlay(alignment: .top) { Rectangle().fill(Theme.border).frame(height: 1) }
                .padding(.top, 12)
                if !isPro {
                    Button { showPro = true } label: {
                        HStack {
                            Text("Unlock your full insights with Pro").font(Brand.font(11, .heavy))
                            Spacer()
                            Image(systemName: "arrow.right").font(.system(size: 11, weight: .bold))
                        }
                        .foregroundStyle(Theme.primary)
                        .padding(.top, 10)
                        .overlay(alignment: .top) { Rectangle().fill(Theme.border).frame(height: 1) }
                    }
                    .buttonStyle(.plain)
                    .padding(.top, 10)
                }
            }
        }
        .sheet(isPresented: $showPro) { ProView() }
    }
}

// MARK: - Daily standing strip

/// "You're in the top X% today · across N dailies" — where today's composite
/// scores sit in the field. Hidden when the user hasn't played a daily today.
struct DailyStandingStrip: View {
    @State private var standing: StatsDeepService.DailyStanding?
    /// Bump to refetch (daily completions change the standing).
    var reloadToken: Int = 0

    var body: some View {
        Group {
            if let s = standing {
                HStack(spacing: 8) {
                    Image(systemName: "chart.line.uptrend.xyaxis").font(.system(size: 14))
                        .foregroundStyle(Theme.primary)
                    (Text("You're in the ").font(Brand.font(11, .heavy)).foregroundColor(Theme.textPrimary)
                     + Text("top \(s.topPercent)%").font(Brand.font(11, .black)).foregroundColor(Theme.primary)
                     + Text(" today").font(Brand.font(11, .heavy)).foregroundColor(Theme.textPrimary)
                     + Text(" · across \(s.modesCounted) \(s.modesCounted == 1 ? "daily" : "dailies")")
                        .font(Brand.font(11, .heavy)).foregroundColor(Theme.textMuted))
                    Spacer(minLength: 0)
                }
                .padding(.horizontal, 16).padding(.vertical, 10)
                .background(RoundedRectangle(cornerRadius: 14).fill(LinearGradient(
                    colors: [Color(hex: 0xF5F3FF), Color(hex: 0xFCE7F3)],
                    startPoint: .topLeading, endPoint: .bottomTrailing)))
                .overlay(RoundedRectangle(cornerRadius: 14).stroke(Color(hex: 0xE9D5FF), lineWidth: 1.5))
            }
        }
        .task(id: reloadToken) { standing = await StatsDeepService.todayDailyStanding() }
    }
}

// MARK: - Opener Lab (basic)

/// Favorite starting words + how they convert (win rate of games opened with
/// each word). Free-tier card — the deep yield version lives in Deep Insights.
struct OpenerLabCard: View {
    /// Play-type scope from the page toggle (restat B1); vs_cpu → empty → hidden.
    var playType: String = "solo"
    @State private var openers: [StatsDeepService.OpenerStat] = []
    @State private var loaded = false

    var body: some View {
        Group {
            if openers.isEmpty, loaded, playType != "vs_cpu" {
                StatsEmptyCard(title: "Opener Lab", accent: Color(hex: 0x06B6D4),
                               hint: "Win a few games and your favorite starting words show up here.")
            } else if !openers.isEmpty {
                VStack(alignment: .leading, spacing: 8) {
                    SectionHeader("Opener Lab", accent: Color(hex: 0x06B6D4))
                    KitCard {
                        VStack(alignment: .leading, spacing: 8) {
                            VStack(spacing: 6) {
                                ForEach(Array(openers.enumerated()), id: \.element.id) { i, o in
                                    HStack(spacing: 10) {
                                        Text("\(i + 1)").font(Brand.font(10, .black))
                                            .foregroundStyle(Theme.textMuted).frame(width: 16)
                                        Text(o.word).font(Brand.font(14, .black)).tracking(1.2)
                                            .foregroundStyle(Theme.textPrimary)
                                        Spacer()
                                        Text("\(o.count)×").font(Brand.font(10, .bold)).foregroundStyle(Theme.textMuted)
                                        Text("\(o.winRate)% W").font(Brand.font(12, .black))
                                            .foregroundStyle(o.winRate >= 50 ? Theme.primary : Color(hex: 0xDC2626))
                                            .frame(width: 48, alignment: .trailing)
                                    }
                                    .padding(8)
                                    .background(RoundedRectangle(cornerRadius: 10).fill(Theme.background))
                                }
                            }
                            Text("Win rate of games opened with each word")
                                .font(Brand.font(9, .bold)).foregroundStyle(Theme.textMuted)
                                .frame(maxWidth: .infinity).multilineTextAlignment(.center)
                        }
                    }
                }
            }
        }
        .task(id: playType) {
            // P-cache: seed from the session memo, then fetch fresh as before.
            let key = "openerLab:\(AuthService.shared.profile?.id ?? "anon"):\(playType)"
            if let cached: [StatsDeepService.OpenerStat] = StatsMemo.shared.get(key) { openers = cached }
            let fresh = await StatsDeepService.openerStats(limit: 5, playType: playType)
            openers = fresh
            loaded = true
            StatsMemo.shared.set(key, fresh)
        }
    }
}

// MARK: - Weekday form

/// Win rate by day of week — highlights your best day (gold bar).
struct WeekdayFormCard: View {
    /// Play-type scope from the page toggle (restat B1); vs_cpu → zero days → hidden.
    var playType: String = "solo"
    @State private var days: [StatsDeepService.WeekdayFormDay] = []
    @State private var loaded = false

    private let labels = ["S", "M", "T", "W", "T", "F", "S"]
    private let dayNames = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]

    private var best: StatsDeepService.WeekdayFormDay? {
        days.filter { $0.played >= 3 }
            .max { Double($0.won) / Double($0.played) < Double($1.won) / Double($1.played) }
    }

    var body: some View {
        Group {
            if !days.contains(where: { $0.played > 0 }), loaded, playType != "vs_cpu" {
                StatsEmptyCard(title: "Weekday Form", accent: Color(hex: 0xF97316),
                               hint: "Play across the week to see your win rate by day.")
            } else if days.contains(where: { $0.played > 0 }) {
                let maxPlayed = max(1, days.map(\.played).max() ?? 1)
                let hint = best.map { "Best: \(dayNames[$0.dow]) (\(Int((Double($0.won) / Double($0.played) * 100).rounded()))%)" }
                VStack(alignment: .leading, spacing: 8) {
                    SectionHeader("Weekday Form", accent: Color(hex: 0xF97316))
                    ChartCard(title: "Win rate by day", hint: hint) {
                        HStack(alignment: .bottom, spacing: 6) {
                            ForEach(days) { d in
                                let rate = d.played > 0 ? Double(d.won) / Double(d.played) : 0
                                VStack(spacing: 4) {
                                    Text(d.played > 0 ? "\(Int((rate * 100).rounded()))%" : " ")
                                        .font(Brand.font(8, .bold)).foregroundStyle(Theme.textMuted)
                                    ZStack(alignment: .bottom) {
                                        Color.clear.frame(height: 44)
                                        RoundedRectangle(cornerRadius: 3)
                                            .fill(d.played == 0 ? AnyShapeStyle(Theme.border)
                                                  : best?.dow == d.dow
                                                  ? AnyShapeStyle(LinearGradient(colors: [Color(hex: 0xFBBF24), Color(hex: 0xF97316)], startPoint: .top, endPoint: .bottom))
                                                  : AnyShapeStyle(LinearGradient(colors: [Color(hex: 0xA78BFA), Color(hex: 0x7C3AED)], startPoint: .top, endPoint: .bottom)))
                                            .frame(height: d.played == 0 ? 2 : 4.4 + rate * 39.6)
                                            .frame(maxWidth: .infinity)
                                            .opacity(d.played == 0 ? 1 : 0.5 + 0.5 * Double(d.played) / Double(maxPlayed))
                                    }
                                    Text(labels[d.dow]).font(Brand.font(9, .heavy)).foregroundStyle(Theme.textMuted)
                                }
                            }
                        }
                    }
                }
            }
        }
        .task(id: playType) {
            let key = "weekdayForm:\(AuthService.shared.profile?.id ?? "anon"):\(playType)"
            if let cached: [StatsDeepService.WeekdayFormDay] = StatsMemo.shared.get(key) { days = cached }
            let fresh = await StatsDeepService.weekdayForm(playType: playType)
            days = fresh
            loaded = true
            StatsMemo.shared.set(key, fresh)
        }
    }
}

// MARK: - Daily points trend

/// Points-per-day line (sweep/flawless days marked) — split out of the old
/// sweep-counts card; the counts moved to Records → You (single home).
struct DailyPointsChartCard: View {
    @State private var points: [MatchStatsService.DailyPointsPoint] = []
    @State private var loaded = false

    var body: some View {
        Group {
            if points.count < 2, loaded {
                StatsEmptyCard(title: "Daily Points", accent: Color(hex: 0xEC4899),
                               hint: "Finish dailies on a few different days to chart your points.")
            } else if points.count >= 2 {
                VStack(alignment: .leading, spacing: 8) {
                    SectionHeader("Daily Points", accent: Color(hex: 0xEC4899))
                    ChartCard(title: "Points per day", hint: "Last 30 days · ● sweep · ● flawless") {
                        Chart {
                            ForEach(points) { p in
                                LineMark(x: .value("Day", p.day), y: .value("Points", p.totalPoints))
                                    .foregroundStyle(Color(hex: 0x7C3AED)).interpolationMethod(.catmullRom)
                                AreaMark(x: .value("Day", p.day), y: .value("Points", p.totalPoints))
                                    .foregroundStyle(LinearGradient(colors: [Color(hex: 0xA78BFA).opacity(0.3), .clear], startPoint: .top, endPoint: .bottom))
                                    .interpolationMethod(.catmullRom)
                                if p.swept || p.flawless {
                                    PointMark(x: .value("Day", p.day), y: .value("Points", p.totalPoints))
                                        .foregroundStyle(p.flawless ? Color(hex: 0xF59E0B) : Color(hex: 0xEC4899))
                                }
                            }
                        }
                        .chartXAxis(.hidden)
                        .frame(height: 110)
                    }
                }
            }
        }
        .task {
            let key = "dailyPoints:\(AuthService.shared.profile?.id ?? "anon")"
            if let cached: [MatchStatsService.DailyPointsPoint] = StatsMemo.shared.get(key) { points = cached }
            let fresh = await MatchStatsService.dailyPointsOverTime(days: 30)
            points = fresh
            loaded = true
            StatsMemo.shared.set(key, fresh)
        }
    }
}
